runtime/reconcilers/model.go

```diff
@@ -1123,6 +1130,14 @@ func (r *ModelReconciler) executePartition(ctx context.Context, catalog drivers.
 		logArgs = append(logArgs, zap.Error(err))
 	}
 
+	// Run partition-level tests after each partition is executed
+	if err == nil && mdl.Spec.PartitionsResolver != "" {
+		testErr := r.runModelTests(ctx, self)
+		if testErr != nil {
+			return nil, false, testErr
+		}
+	}
```

Comment:
If you want to support partition-level tests, it'd need to pass the `data` variable in here and make it available in templating. Otherwise checks like `where uri = '{{.partition.uri}}'` wouldn't work.

The problem then is that if partitions can be referenced, then it will only work if partition context is provided. I wonder if a separate `partition_tests:` is the only option or if we can think of another way.